internal/gouging/gouging.go

```diff
@@ -221,15 +210,6 @@ func checkPriceGougingPT(gs api.GougingSettings, cs api.ConsensusState, pt *rhpv
 		if pt.HostBlockHeight < cs.BlockHeight {
 			return fmt.Errorf("consensus not synced and host block height is lower, %v < %v", pt.HostBlockHeight, cs.BlockHeight)
 		}
-	} else {
-		var minHeight uint64
```

Comment:
We can hardcode the leeway to 12 blocks but I don't think we can remove the check. Nothing protects us from overpaying on uploads without it. 
For that reason, the block height is one of the relatively few fields of the price table that made it into rhp4 prices as well.

Reply:
Another thing to keep in mind is that the v2 feature branch heavily changes gouging settings. So any change here will introduce massive conflicts. So we might be better off refactoring certain less urgent parts after the v2 code is merged.

Reply:
I hardcoded a default of 12. The updated PR removes a function and swaps a setting for a default, so hopefully that won't massively conflict... I understand that would be annoying though, I'll leave it up to you on whether you think it'll be too much of a pain. Getting rid of this setting allows us to easily get rid of `WithGougingChecker` which makes it easy to fetch gouging parameters on the fly and only use the cache for usable hosts.

Reply:
It would indeed conflict on [this update](https://github.com/SiaFoundation/renterd/pull/1418/files#diff-eb06058a1f2ee16c08a6db3d0cf6be2125e6cc385ccd2eb118d9df982a719c92R283), since its-happening is very up-to-date with `dev` the merge wouldn't be that much of a pain